pkcs12/src/kdf.rs

```diff
@@ -0,0 +1,110 @@
+//! Implementation of the key derivation function
+//! [RFC 7292 Appendix B](https://datatracker.ietf.org/doc/html/rfc7292#appendix-B)
+
+use alloc::vec::Vec;
+use digest::{core_api::BlockSizeUser, Digest, FixedOutputReset, OutputSizeUser, Update};
+use zeroize::Zeroize;
+
+/// Transform a utf-8 string in a unicode (utf16) string as binary array.
+/// The Utf16 code points are stored in big endian format with two trailing zero bytes.
+pub fn str_to_unicode(utf8_str: &str) -> Vec<u8> {
```

Comment:
I don't think this needs to be `pub`?

```suggestion
fn str_to_unicode(utf8_str: &str) -> Vec<u8> {
```